internal/name/name.go

```diff
@@ -29,20 +29,22 @@ func Limit(s string, count int) string {
 		return s
 	}
 
-	if count <= 6 {
+	hexLen := 5
+	separator := "-"
+
+	if count <= hexLen+len(separator) {
 		return s[:count]
 	}
 
-	separator := "-"
-	nbCharsBeforeTrim := count - 6
+	nbCharsBeforeTrim := count - hexLen - 1 // using a 1-char separator
 
 	// If the last character of the truncated string is the separator, include it instead of the separator.
 	if string(s[nbCharsBeforeTrim-1]) == separator {
 		separator = ""
-		nbCharsBeforeTrim++
+		hexLen++
 	}
 
-	return fmt.Sprintf("%s%s%s", s[:nbCharsBeforeTrim], separator, Hex(s, 5))
+	return fmt.Sprintf("%s%s%s", s[:nbCharsBeforeTrim], separator, Hex(s, hexLen))
```

Comment:
The issue with #1647 is that the resulting hash may contain non-hex characters. This is a cosmetics rather than functional issue: by eliminating double dashes, #1647 already takes care of preventing inconsistencies between successive calls of `HelmReleaseName`, and therefore between bundle ID and Helm release names.

In any case, increasing the hex length is perhaps not the best approach to this. Let's simply remove the extra dash if any, as discussed. The only caveat is that the result (ie bundle ID/Helm release name) will contain one character fewer than the limit in such cases, but that should not be an issue.